app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+  validates :name, presence: true
+  validates :email, uniqueness: true
+  has_secure_password
```

Comment:
giải thích line code này có tác dụng gì nhé

Reply:
before_save :downcase_email -> trước khi 1 bản ghi của bảng User được lưu vào Database thì email phải được  biến đổi thành chuỗi thường
validates :name, presence: true -> không cho phép name của User là nil hoặc là xâu rỗng hoặc chứa toàn các kí tự trắng
validates :email, uniqueness: true -> email phải là duy nhất trong bảng
has_secure_password -> lưu mật khẩu dưới mã hash, tạo 2 thuộc tính ảo là password và password_confirmation, có phương thức authenticate (cung cấp mật khẩu) đế xác thực người dùng